easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,802 @@
+
+    def build_user_molfile(self, verbose=False, path=None):
+        """
+        Build the user-molfile package
+        """
+        raise EasyBuildError("No instructions yet for building the user-molfile package.")
+
+    def build_user_omp(self, verbose=False, path=None):
+        """
+        Build the user-omp package
+        """
+        os.chdir(os.path.join(self.cfg['start_dir'], 'src'))
+        cmd = "make yes-user-omp"
+        (user_omp_src_output, _) = run_cmd(cmd, path=path, log_all=True, simple=False, log_output=verbose)
+        return user_omp_src_output
+
+    def build_user_phonon(self, verbose=False, path=None):
+        """
+        Build the user-phonon package
+        """
+        raise EasyBuildError("No instructions yet for building the user-phonon package.")
+
+    def build_user_qmmm(self, verbose=False, path=None):
+        """
+        Build the user-qmmm package
+        """
+        raise EasyBuildError("No instructions yet for building the user-qmmm package.")
+
+    def build_user_qtb(self, verbose=False, path=None):
+        """
+        Build the user-qtb package
+        """
+        raise EasyBuildError("No instructions yet for building the user-qtb package.")
+
+    def build_user_quip(self, verbose=False, path=None):
+        """
+        Build the user-quip package
+        """
+        raise EasyBuildError("No instructions yet for building the user-quip package.")
+
+    def build_user_reaxc(self, verbose=False, path=None):
+        """
+        Build the user-reaxc package
+        """
+        raise EasyBuildError("No instructions yet for building the user-reaxc package.")
+
+    def build_user_smd(self, verbose=False, path=None):
+        """
+        Build the user-smd package
+        """
+        raise EasyBuildError("No instructions yet for building the user-smd package.")
+
+    def build_user_smtbq(self, verbose=False, path=None):
+        """
+        Build the user-smtbq package
+        """
+        raise EasyBuildError("No instructions yet for building the user-smtbq package.")
+
+    def build_user_sph(self, verbose=False, path=None):
+        """
+        Build the user-sph package
+        """
+        raise EasyBuildError("No instructions yet for building the user-sph package.")
+
+    def build_user_tally(self, verbose=False, path=None):
+        """
+        Build the user-tally package
+        """
+        raise EasyBuildError("No instructions yet for building the user-tally package.")
+
+    def build_step(self, verbose=False, path=None):
+        """
+        Start the actual build
+        - typical: make -j X
+        """
+
+        # Can't be None as needs to be appended to
+        out = ''
+        src_dir = os.path.join(self.cfg['start_dir'], 'src')
+
+        if self.toolchain.options.get('openmp', None):
+            out += self.build_user_omp(verbose, path)
+
+        if self.cfg['use_asphere']:
+            out += self.build_asphere(verbose, path)
```

Comment:
this can be collapsed with some more Python magic...

```python
for comp in components:
    if self.cfg['use_%s' % comp]:
        getattr(self, 'build_%s' % comp)(verbose, path)
```

(with some more checking)